src/Pages/GlobalConfigurations/Authorization/UserAndGroupPermissions.tsx

```diff
@@ -0,0 +1,277 @@
+import { mainContext } from '../../../components/common/navigation/NavigationRoutes'
+import { getJobs } from '../../../components/Jobs/Service'
+import { getProjectList } from '../../../components/project/service'
+import { EntityTypes } from './shared/components/userGroups/userGroups.types'
+import { ACCESS_TYPE_MAP, API_STATUS_CODES, Routes, SERVER_MODE } from '../../../config'
+import { getEnvironmentListHelmApps, getEnvironmentListMin, getProjectFilteredApps } from '../../../services/service'
+import './authorization.scss'
+import { getCustomRoles, getPermissionGroupList } from './authorization.service'
+import { AuthorizationProvider } from './AuthorizationProvider'
+import { getMetaPossibleRoles } from './utils'
+import { UserAndGroupPermissionsWrapProps } from './types'
+
+const APITokens = lazy(() => import('./APITokens'))
+const UserPermissions = lazy(() => import('./UserPermissions'))
+const PermissionGroups = lazy(() => import('./PermissionGroups'))
+
+const AuthorizationGlobalConfigWrapper = importComponentFromFELibrary('AuthorizationGlobalConfigWrapper')
+
+const UserAndGroupPermissionsWrap = ({ children, setIsAutoAssignFlowEnabled }: UserAndGroupPermissionsWrapProps) => {
+    const getWrap = (child) => (
+        <AuthorizationGlobalConfigWrapper setIsAutoAssignFlowEnabled={setIsAutoAssignFlowEnabled}>
+            {child}
+        </AuthorizationGlobalConfigWrapper>
+    )
+
+    return (
+        <ConditionalWrap condition={!!AuthorizationGlobalConfigWrapper} wrap={getWrap}>
+            {children}
+        </ConditionalWrap>
+    )
+}
+
+const UserAndGroupPermissions = () => {
+    const { serverMode } = useContext(mainContext)
+    const { path } = useRouteMatch()
+    const [isDataLoading, data, error, reload] = useAsync(() =>
+        Promise.all([
+            // TODO (v3): Remove this from here and use type ahead search instead
+            getPermissionGroupList({
+                showAll: true,
+            }),
+            getProjectList(),
+            getEnvironmentListMin(),
+            serverMode === SERVER_MODE.EA_ONLY ? null : getChartGroups(),
+            getEnvironmentListHelmApps(),
+            getCustomRoles(),
+        ]),
+    )
+    // TODO (v3): use object instead and try to get rid of this altogether
+    const [appsList, setAppsList] = useState(new Map())
+    const [appsListHelmApps, setAppsListHelmApps] = useState(new Map())
+    const [jobsList, setJobsList] = useState(new Map())
+    // For handling the auto assign flow for enterprise
+    const [isAutoAssignFlowEnabled, setIsAutoAssignFlowEnabled] = useState(false)
+
+    // TODO (v3): possibly move to a provider
+    async function fetchJobsList(projectIds: number[]) {
+        const missingProjects = projectIds.filter((projectId) => !jobsList.has(projectId))
+        if (missingProjects.length === 0) {
+            return
+        }
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+        setJobsList((jobsList) => {
```

Comment:
Why are we setting this here? it is anyways getting handled in try and catch both

Reply:
will revisit in v2